models/schemas/achievementSchema.ts

```diff
@@ -0,0 +1,21 @@
+import mongoose, { Mongoose } from 'mongoose';
+import AchievementType from '../../constants/achievementType';
+import Achievement from '../db/achievement';
+
+
+const AchievementSchema = new mongoose.Schema<Achievement>({
+    type: {
+        type: String,
+        enum: AchievementType,
+        required: true,
+    },
+    userId: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: 'User',
+        required: true,
+    },
+}, {
+    timestamps: true,
+});
+
```

Comment:
Probably should have index for type and userId. Also maybe that index should be unique

Reply:
Can do this instead of `index: true`:

`AchievementSchema.index({ type: 1, userId: 1 }, { unique: true });`